File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/events/SLInventoryNewContentsEvent.java
package com.lumiyaviewer.lumiya.slproto.events;

import java.util.UUID;

public class SLInventoryNewContentsEvent {
    public boolean firstIsFolder = false;
    public String firstItemName = null;
    public UUID firstParentUUID = null;
    public int newFolderCount = 0;
    public int newItemCount = 0;

    public void AddItem(boolean z, UUID uuid, String str) {
        if (z) {
            this.newFolderCount++;
        } else {
            this.newItemCount++;
        }
        if (this.firstParentUUID == null) {
            this.firstIsFolder = z;
            this.firstParentUUID = uuid;
            this.firstItemName = str;
        } else if (z && (!this.firstIsFolder)) {
            this.firstIsFolder = z;
            this.firstParentUUID = uuid;
            this.firstItemName = str;
        }
    }

    public boolean isEmpty() {
        return this.newFolderCount == 0 && this.newItemCount == 0;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/events/SLInventoryNewContentsEvent.kt
package com.lumiyaviewer.lumiya.slproto.events

import java.util.UUID

class SLInventoryNewContentsEvent {
    var firstIsFolder = false
    var firstItemName: String? = null
    var firstParentUUID: UUID? = null
    var newFolderCount = 0
    var newItemCount = 0
    
    fun AddItem(isFolder: Boolean, parentUUID: UUID, itemName: String) {
        if (isFolder) {
            newFolderCount++
        } else {
            newItemCount++
        }
        
        if (firstParentUUID == null) {
            firstIsFolder = isFolder
            firstParentUUID = parentUUID
            firstItemName = itemName
        } else if (isFolder && !firstIsFolder) {
            firstIsFolder = isFolder
            firstParentUUID = parentUUID
            firstItemName = itemName
        }
    }
    
    fun isEmpty(): Boolean = newFolderCount == 0 && newItemCount == 0
}